### PR TITLE
Add system-wide configuration file and installtion.

### LINKED ISSUE #62: Add system-wide configuration file
`/etc/debops/debops.cfg` should be used as the default system-wide configuration file. It could specify an address and branch of default `debops-playbooks` git repository, which could allow to easily install playbook and roles from different repositories instead of the official one.

### LINKED ISSUE #85: Debops system-wide installation option(s)
DebOps used to do a system-wide installation in to /usr/local.  Since v0.2.0, it no longer does and instead installs in $HOME/.local.  This might work well when you have a single person working with DebOps, but once you have more than one, the system-wide installation is required.

A suggestion I have is to have debops default to using a global /etc/debops.conf file whose location can be overridden by an environment variable, or a command line arg... or both.  In there you can set what the prefix is for the installation path.  That way you can support user and system-wide installations.

### PR DESCRIPTION
**Pull request by htgoebel**
Closes #85, closes #62. Defaults are:

```
[paths]
data-home: $XDG_CONFIG_HOME/debops

# Default installation directory
install-path: %(data-home)s/debops-playbooks

# Locations where DebOps playbooks might be found
# This MUST be a multi-line string to make ConfigParser work
playbooks-paths: %(install-path)s/playbooks
```

### CODE REVIEW

**Comment by htgoebel** [comment]
@drybjed please hold on, it is not yet finished, I discovered bugs.

File: debops/config.py [hunk 1]
```diff
@@ -37,6 +38,29 @@
...
+# Default installation directory
+install-path: %(data-home)s/debops-playbooks
+
+# Locations where DebOps playbooks might be found
+# This MUST be a multi-line string to make ConfigParser work
+playbooks-paths: %(install-path)s/playbooks
+"""
+
+def _set_xdg_defaults():
+    """
```
Comment (do3cc) [inline debops/config.py:62]: Currently I need to work with absolute paths in some cases. XDG Paths are OS dependent. So I could only write my code stuff workable for one OS.
For example, my file debops_dir/ansible/playbooks/site.yml:

```

- include: ...
- include: ~/.local/share/debops/debops-playbooks/playbooks/site.yml
```

Comment (htgoebel) [inline debops/config.py:62]: What exactly do you want me to change?

Comment (do3cc) [inline debops/config.py:62]: OS independent default paths

Comment (htgoebel) [inline debops/config.py:62]: What _exactly_ should I change? How should 'OS independent default paths' be done?

Comment (drybjed) [inline debops/config.py:62]: These probably should check what OS the script is executed in and use relevant config variables for default paths? I'm not sure what's the `XDG_*` equivalent on MacOSX though...

Comment (do3cc) [inline debops/config.py:62]: ~/something would be os independent

Comment (htgoebel) [inline debops/config.py:62]: Simply tell me what I should put there :-)
Maybe this is helpful: http://stackoverflow.com/questions/3373948/equivalents-of-xdg-config-home-and-xdg-data-home-on-mac-os-x
@do3cc If you are on different platforms, you should be able to look where other applications store their data.

Comment (do3cc) [inline debops/config.py:62]: -----BEGIN PGP MESSAGE-----
Version: APG v1.1.1

hQEMAxHP1UhKbkN1AQf7BXJLc7YqdABinoSr78Dp8QP4XiW4g2OxqAYjwqqlFVjq
qCgKCpziVoq7jIvED9FemZWOwnq3srVEesmL1ZdQoax4xHaRpKcZjgmR3pRPOwHN
a0/fLHm2R2yjwD5ocUdRJh4KXdcbBVmyVCTG4X2UyOXUtY3THYbziSYB04Wzopu6
hIZRFIMxy6eYakZZahiCf2WGhLJK+ZZ6ytbZx83rwNm4L2pw52nW5O9lsi9/+l2a
6wHr4Fw9kJJyRNqpEXtSfVM4TZ21kyABsq9Uwm9N6NGxvg4J7WHTIWAJXm0wVqcl
/IEnsSZ8gv9H4Z9ewV2Y4/ZSc3eTTw1nscLVf44wwYUCDAP3AmoPMC4ZOwEQAIRq
Mbc2EKpaiJ3S0MfICbHR+YAGB0X23gMrqTAqBCZvwDraGtH7dziBzV/NCYPWlcF0
M48mLp+ItPbUIXH04v2sMXdV+tCii3C4BPdd+yg6JrBZmWmD3ORLBGBpg9fmO/d+
tFTCKfz50KdDo3cRxbE85o45cBZcpSTg2z71U0ZVuyMvbe9taHn8OvaLSOwrG+vG
Qo5ZepbPC2loBridd9ViA4/O+dhJ0pLc8xq7DqEik7R4aKtv+K/MQAys28wnM/su
fd7IiVxDzy3z8PPSQlzFXF/1sM3i1vBngTNSmyqeEUrfKraZiLbOSKmOUOiDNWjB
ZuZ5NGSc+WpgMHz6eSmidy7AKHTzrYEc8d36CMMnOlUBtSqnBr5nglQlkSq4R7CC
JGpihXO03APdYUSVqU3cjirZYGNvjJ7gzYzGhZZzcgtIHpU/784ET7C/9QFHdabv
hIQCcVcTSFqO457fQXq57At4yH23utUvO7JDDojsm4+3i7VfMKd4VoYuNTSFd8NU
s80v7SLRGVtRvGBkH8oY8dSC4RRUbfe2DR59bu+DMTL7+N/+OnH5cQmskV4lQyvz
8ftUGVDXtlC81aH/xYJvPY78jgsHYkajgjLd4VqtmG4UQ2tyZaC9AE9ScUbLzCQt
7JM41bfuY+M8FBXLqozRoGyZomsuEWa+6ckydXpZ0sWrASGgA6k+CdK9TAJ4JD/n
bTBfhc9eKtu9PTt6IzxlpKWppE3TIAnzv5czTdrJa7NIlJjIPgkS7OfsyzPz1/R6
2syFS8ndeaOa32w/pLaO8ZY2nzYvqzzXnMuSQyvSSY/CtqChXs/x+wDLXz7ix4ar
9CvW3AVEtkXiE0bxgsm4g1QgJpFV4OjV8iRV44ASiKG6W7ZnjKaGJdjbm1xFeNrW
PSF/a6v+r6szPpD27u3/GHEZAezTvfHVtSMyMuJmoGH2Lwz9MeYLDug/Vlw1ZdO7
T1M7ZjOHw4tYMVbKVKDLZwBF/mC6BFYRGIlkOskZC+tk/YKjakdNFAhWM3NkaBSd
TOZIyyXjKtIk/ruo7i0bExD6yO8qU+cXoR7fPKae06YLYE4DrgujkMnR9MjFDDvc
m5JnU0wDbIW5RmLPcRW/zCjzjSZ2DbGFzVCRpO0aJZUYgTfuaSnL5SMM6bWy8OP/
W9ZRosuP09dF+XzLLtcJeXkLFHL8Nawmo73wn+7kAmj0baXftgOpx1JhpeqXryJY
DGJ20w2qAzJSzoEe93JWfCaEtcyW+be9BlTZJbq9tg8nPim26ExhoXUor0c8UZTP
/P4ug5Ope0sdjrZd0dnloggZ8XeLYXjupZg+x5OzW0M9HZNQ1YHKABuiTNkwLahJ
j/1yjcVxjdG4vI60VsB+rDt30hbPVT9yDCO/87iR8S2gAzFf2GiNOOTtGxnLwiAU
+FmOpCrotQSvbOcItlYq8DxWqLV9DhuRx02VthJfYwwrTioQeIuIsr+7ChzX7Qpo
BLiaUTLsK7yHuJVdCKGvtP0PRgqj/X13dz+COih+5cOrTLeo3V5ZXR73TtqRfjYJ
F7dKrMccXcDT7jkmUbbzOHXhZsTXNyHuYlUoayVs6s2k4Gjxn81RQfYVvk2k9cop
rwqsRmo99rspqrbS0oCOfe+r7D36QX2xgVcgcxhDO2b7PQi3ZpzaxM4t14XxqK65
HM86SXj5uRNS4rNm1b0NZ9dj3bK05EEW31sr2LyrTxCrkhH4rKR2C32MDrmldJb/
zWyyz0Au/KKEwq8/LY8oGNgPHJKop9CjLnWB08VGI+ytAAvcMCjtLbSMV4ixoeWY
W9a0ohuphPgxvzKIn21fhNbb83VhB/iA72Ke3KQzyUyvyydw/YBMGZWSFpiYhNqn
ijol9n1ORdN1X7kPFvXBH7hgr6sfBeD5IS/GYz9JY01Pd647BIL6XgnRnamFbfp/
1L6bZu0trs/FWKiaZKB1XEdXYz9kVHYQAcYisQ4WwGnX2BAb7p+GV/KRapa7+l06
/kNTm8qK967ARy0nmakux6Jhe7Pm3d6xRxQXSjXjzfqvzUh3sOki6fiJP2GYyNcp
w361bzuBlNbDVdFJSzmMN/RIkx1hoJ8tZSS4jytSj/HoCC319EyvxtoLlwBxy2yp
Ck7JK/Cpd35F7XukeR5ZO8zqC0ZLH7zhYvw8xRJJLPEQ4RKmGD7wejtme07+Tt4S
dPUCPH1qpP20kNhuX6TZimtWyck4/i14TKpTJfrHX76ChkC76fcEQUyA4Zj3qDWa
NhK/eicnnEE0dYhUKXEfekguPEhPoQUsnI9juMPVidhcNQLZLEdnkfqyh3ev4F2w
DfCcH5g+5Z3QSDssZB67HFthDyNICwekL0+cjgz9G9Y84TnUEzGw/LGEff0pCni3
c90lNCLMKaAzPz6BSxBSNjCzzEnk8saGaI6nzI0eEl7E8AJpSgK0PFtJzolHhuax
zmNwVh2rbCBwB0NKnRtEKKyBLS4o084O+MzjZeib2+aqxTjSjJlrPBRP5uExQyWn
rd4uKjqnWi7bUWRIi1RLYIXWb3DEsiLEEXg1r0m7mquGrisSNDfkIxQYieC44n29
rZ6zGBmXZTUPgbEZZKnnJ2re86jGrlFiynOeh9WJpZ205szMpONlVV908bmxLIdD
xJgEObUaalaXUW5BgkX60glxqv8PwsI5j0F0gZcpogFIP+Wd7D0SCuXmC7SG5+3B
kdv8D0FGPxFq9ZqxY7vt6DHhWemlwsAhyvUqIKykTNIbXG3wi61o/4iy5MRQUqF6
NgPMjmrpdCR2JpdXTUSUMCg+ICfO/TGoQ5tIQ2prdslfV/+l8uhzXyTT4t0RR3zC
KoPEfjPmetGW47fM0QzlDy/J46MZII3XWdYIERSyqAfllOLNXZEkQum8jjU=
=hI4g
-----END PGP MESSAGE-----

Comment (do3cc) [inline debops/config.py:62]: Since nobody replied, I guess everybody can agree on my last comment

Comment (do3cc) [inline debops/config.py:62]: Ok, here the unencrypted comment:
We are clearly miscommunicating, but for today i won't be able to write much more. (This was yestereday)
`os.path.expand("~/debops")` would be the right answer of course. This way my example would work for my coworker still not feed up with his Mac and me.
I know how to find out where the default values for xdg something are stored on each os. There isa python package for getting just this info. But i cantrun python in my site.yml

Comment (htgoebel) [inline debops/config.py:62]: `~/debops` is wrong for all platform, so I will not make this the default.

**Comment by htgoebel** [comment]
@drybjed Okay, bugs are fixed (hopefully :-) and platform-specific default paths are implemented. Will update the docs soon.

**Comment by drybjed** [comment]
@htgoebel Thanks! I'll let that code be tested a little before releasing the new version on PyPI.

**Pull request merged**